src/main/java/co/tryterra/terraclient/models/AuthData.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2022 Terra Enabling Developers Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package co.tryterra.terraclient.models;
+
+import co.tryterra.terraclient.api.User;
+import lombok.Data;
+
+@Data
+public class AuthData {
+    private final User user;
+    private final String reference_id;
+    private final String widget_session_id;
```

Comment:
You shouldn't use underscores in field names anywhere. If this is being created from a JSON payload with underscores you can use: 
```java
@JsonProperty("field_name_with_underscores")
private final ... fieldNameWithUnderscores;
```
You may also want to be careful using final fields as I believe jackson uses setters to set the fields when deserializing. In the other models I omit the `final` modifier to prevent an exception being thrown.

Reply:
In addition to this, you can see that in other models I use a larger annotation stack to modify behaviours:
```java
@Data  // adds getters
@NoArgsConstructor  // empty constructor as jackson uses reflection to set values (not setters I was wrong above)
@Setter(AccessLevel.NONE)  // remove the setters added by @Data
@JsonIgnoreProperties(ignoreUnknown = true)  // make sure deserialization doesn't fail if unrecognised fields are added
```

Reply:
Re private convo: class is being constructed manually so we can ignore the final part